lms/djangoapps/course_blocks/transformers/user_partitions.py

```diff
@@ -48,7 +49,7 @@ def collect(cls, block_structure):
         root_block = block_structure.get_xblock(block_structure.root_block_usage_key)
         user_partitions = [
             user_partition
-            for user_partition in getattr(root_block, 'user_partitions', [])
+            for user_partition in get_course_user_partitions(root_block)
```

Comment:
@nasthagiri transformer code change. I am assuming "root_block" here is the course.